### ISSUE TITLE
[BUG]error augmented when use deepspeed zero3

### ISSUE THREAD

**Issue by duanzhenyu001** [microsoft/DeepSpeed#4987]
Hi, when i use zero stage 3 to load an yi-6b model and use it like this
"""
model_path = "/mnt/bn/mods-llm/models_collects/Yi-6B-Chat"
args = add_argument()
training_args = TrainingArguments(deepspeed='ds_config.json', output_dir="./")
with zero.Init():
    yi_model = AutoModelForCausalLM.from_pretrained(model_path, torch_dtype=torch.float16)

print(f"model loaded!")
parameters = filter(lambda p: p.requires_grad, yi_model.parameters())
ds_engine, optimizer, _, _ = deepspeed.initialize(args=args, model=yi_model.model, model_parameters=parameters)
ds_engine.eval()
tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=False)
inputs = tokenizer(["我们都有一个家，名字叫中国，兄弟姐妹"])
inputs["input_ids"] = inputs["input_ids"].to(ds_engine.device)
inputs["attention_mask"] = inputs["attention_mask"].to(ds_engine.device)
predicts = ds_engine(**inputs, output_hidden_states=True)
predicts = predicts.logits
out_id = predicts[0][-1].argmax(keepdim=True).tolist()
print(f"predicts shape: {predicts.shape}")
proba = predicts[0][-1].max()
token = tokenizer.convert_ids_to_tokens(out_id)
print(f"new moe model out_id: {out_id}, token: {token} proba: {proba} \n", flush=True)
"""
i got an unexpected output token '兄弟' and proba is 15.375 , different from the output when i use zero1 / zero2 or don't use deepspeed which output is '都' and proba are all 20.328. 
and when i print the models middle output and compered with origin model i found that it was first sudden deviation after post layernorm operator and then augmented after mlp layer.
wondering why this happened, quantization communication？ and how can i decrease this deviation. 
thank very much!

**Comment by duanzhenyu001** [comment]
it seems to be the env problem, when i changed to another torch+cuda version this problem resloved
